src/features/notes/components/Notes.js

```diff
@@ -81,11 +189,70 @@ const Notes = () => {
         <PageWrapper>{`${page + 1}`}</PageWrapper>
         <IconWrapper
           onClick={() => nextPage()}
-          disabled={page >= Math.floor(totalCount / limit)}
+          disabled={!(page < Math.floor(totalCount / limit))}
         >
           <ArrowRightCircle />
         </IconWrapper>
       </Wrapper>
+      <Modal
```

Comment:
- I would separate all this logic regarding the Modal content into a separate file and pass the props like `isOpen`, `onClose`, etc. to that component. This way, you are making smaller parts of code that are much easy to maintain.